server/subscription.go

```diff
@@ -37,7 +50,46 @@ func (s *Server) registerSubscriptionRoutes(g *echo.Group) {
 			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to store license").SetInternal(err)
 		}
 
-		s.subscription = s.loadSubscription()
+		ctx := c.Request().Context()
+		s.subscription = s.loadSubscription(ctx)
+
+		c.Response().Header().Set(echo.HeaderContentType, echo.MIMEApplicationJSONCharsetUTF8)
+		if err := jsonapi.MarshalPayload(c.Response().Writer, &s.subscription); err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to marshal subscription response").SetInternal(err)
+		}
+		return nil
+	})
+
+	g.POST("/subscription/trial", func(c echo.Context) error {
+		license := &enterpriseAPI.License{
```

Comment:
Shall we return error if trial is already setup?

Reply:
Update. We will use the return value for `CreateSettingIfNotExist`, and compare the `IssuedTs` field to check if already exists.

Reply:
Shall we use FindSetting() instead?

Reply:
We’re using `FindSetting` inside the `CreateSettingIfNotExist`, so maybe another `FindSetting` is duplicated and makes it slower.